tests/chainer_tests/functions_tests/rnn_tests/test_function_lstm.py

```diff
@@ -29,151 +28,112 @@ def inject_backend_tests(method_names):
-
-    def test_flat_full_backward(self, backend_config):
-        self.check_backward(
-            self.flat(self.inputs), self.flat(self.grad_outputs),
-            backend_config)
-
-    def test_no_gc_backward(self, backend_config):
-        grad_outputs = [None, self.grad_outputs[1]]
-        self.check_backward(self.inputs, grad_outputs, backend_config)
-
-    def test_flat_no_gc_backward(self, backend_config):
-        grad_outputs = [None, self.grad_outputs[1]]
-        self.check_backward(
-            self.flat(self.inputs), self.flat(grad_outputs), backend_config)
-
-    def test_no_gh_backward(self, backend_config):
-        grad_outputs = [self.grad_outputs[0], None]
-        self.check_backward(self.inputs, grad_outputs, backend_config)
-
-    def test_flat_no_gh_backward(self, backend_config):
-        grad_outputs = [self.grad_outputs[0], None]
-        self.check_backward(
-            self.flat(self.inputs), self.flat(grad_outputs), backend_config)
-
-    def check_double_backward(
-            self, inputs, grad_outputs, grad_grad_inputs, backend_config):
-        if backend_config.use_cuda:
-            inputs = cuda.to_gpu(inputs)
-            grad_outputs = cuda.to_gpu(grad_outputs)
-            grad_grad_inputs = cuda.to_gpu(grad_grad_inputs)
-
-        with backend_config:
-            gradient_check.check_double_backward(
-                chainer.functions.lstm, inputs, grad_outputs, grad_grad_inputs,
-                **self.check_double_backward_options)
-
-    def test_double_backward(self, backend_config):
-        self.check_double_backward(
-            self.inputs, self.grad_outputs, self.grad_grad_inputs,
-            backend_config)
+        def _extract_gates(x):
+            r = x.reshape((len(x), x.shape[1] // 4, 4) + x.shape[2:])
+            return [r[:, :, i] for i in six.moves.range(4)]
+        a, i, f, o = _extract_gates(x)
+        a = numpy.tanh(a)
+        i = sigmoid(i)
+        f = sigmoid(f)
+        o = sigmoid(o)
+        c_exp = numpy.zeros_like(c)
+        c_exp[:batch] = a * i + f * c[:batch]
+        h_exp = o * numpy.tanh(c_exp[:batch])
+        c_exp[batch:] = c[batch:]
+        return c_exp, h_exp,
+
+    def generate_grad_outputs(self, outputs_template):
+        grad_out = []
+        c = outputs_template[0]
+        h = outputs_template[1]
+
+        c_shape = c.shape
+        h_shape = h.shape
+        if self.grad_outputs[0] is True:
+            grad_out.append(numpy.random.uniform(-1, 1,
+                                                 c_shape).astype(c.dtype))
+        else:
+            grad_out.append(None)
+
+        if self.grad_outputs[1] is True:
+            grad_out.append(numpy.random.uniform(-1, 1,
+                                                 h_shape).astype(h.dtype))
+        else:
+            grad_out.append(None)
+        return tuple(grad_out)
+
+    def _generate_grad_outputs(self, outputs_template):
```

Comment:
Do not override `FunctionTestCase._generate_grad_outputs` because this method is defined as private method.

Reply:
`FunctionTestCase._generate_grad_outputs` raises an error for `None` inputs, hence I had to change it to handle `None` inputs. This is currently being fixed in #7831.

Reply:
#7831 has been merged.